File: modules/agent/src/models/IndividualRelationship.py
from sqlalchemy import Column, Text, DateTime, BigInteger, ForeignKey, func
from sqlalchemy.orm import relationship
from models import Base


class IndividualRelationship(Base):
    """Relationship between two individuals."""

    __tablename__ = "Individual_Relationship"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    from_individual_id = Column(BigInteger, ForeignKey("Individual.id", ondelete="CASCADE"), nullable=False)
    to_individual_id = Column(BigInteger, ForeignKey("Individual.id", ondelete="CASCADE"), nullable=False)
    relationship_type = Column(Text, nullable=True)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    from_individual = relationship("Individual", foreign_keys=[from_individual_id], backref="outgoing_relationships")
    to_individual = relationship("Individual", foreign_keys=[to_individual_id], backref="incoming_relationships")

File: modules/agent-go/internal/models/IndividualRelationship.go
package models

import "time"

type IndividualRelationship struct {
	ID               int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	FromIndividualID int64     `gorm:"not null;index" json:"from_individual_id"`
	ToIndividualID   int64     `gorm:"not null;index" json:"to_individual_id"`
	RelationshipType *string   `json:"relationship_type"`
	Notes            *string   `gorm:"type:text" json:"notes"`
	CreatedAt        time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt        time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (IndividualRelationship) TableName() string {
	return "Individual_Relationship"
}
